lib/git-environment.ts

```diff
@@ -29,16 +32,23 @@ function resolveGitBinary(): string {
 
 /**
  * Find the path to the embedded git exec path.
+ *
+ * If a custom git exec path is given as the `GIT_EXEC_PATH` environment variable,
+ * then it returns with it after resolving it as a path.
  */
 function resolveGitExecPath(): string {
-  const gitDir = resolveGitDir()
-  if (process.platform === 'darwin' || process.platform === 'linux') {
-    return path.join(gitDir, 'libexec', 'git-core')
-  } else if (process.platform === 'win32') {
-    return path.join(gitDir, 'mingw64', 'libexec', 'git-core')
-  }
+  if (process.env.GIT_EXEC_PATH) {
+    return path.resolve(process.env.GIT_EXEC_PATH)
+  } else {
```

Comment:
To reduce indenting, you can avoid the `else` clause here as it's unnecessary...